### PR TITLE
tutorial vignette (SIR model)

### PR DESCRIPTION
**Pull request by slwu89**
This PR adds a new vignette "Tutorial.Rmd" that takes users through a SIR model using processes, events, and render objects to show off the capabilities of the package. I'm not sure how the plot it produces displays on all monitors, I'd appreciate it if that could be checked (it looks like it has a fine resolution/size on my machine).

### CODE REVIEW

File: vignettes/Tutorial.Rmd [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+```{r}
+recovery_event <- TargetedEvent$new(population_size = N)
+recovery_event$add_listener(function(t, target) {
+  health$queue_update("R", target)
+})
+```
+
+Finally, we need to define a recovery process that queues future recovery events. We first get `Bitset` objects of those currently infectious individuals and those who have already been scheduled for a recovery. Then, using bitwise operations, we get the intersection of already infectious persons with persons who have not been scheduled, precisely those who need to have recovery times sampled and recoveries scheduled. We sample those times from `rgeom`, where the probability for recovery is $1-e^{-\gamma \Delta t}$. Note that we add one to the resulting vector, because by default R uses a "number of failures" parameterization rather than "number of trials", meaning it would be possible for an individual to be infectious for 0 time steps without the correction. Finally we schedule the recovery using the recovery event object.
```
[the remark this answers is not in the document]
Comment (slwu89) [reply vignettes/Tutorial.Rmd:77]: Not sure how to do this other than a direct markdown link like `[Bitset](https://mrc-ide.github.io/individual/reference/Bitset.html)`, is there a better way that can use internal links?

[the remark this answers is not in the document]
Comment (slwu89) [reply vignettes/Tutorial.Rmd:77]: Ah found it. Can do `individual::Bitset` and that makes nice links in pkgdown.

**Comment by slwu89** [comment]
@giovannic I think this last push addresses all your comments (I added nice linking between vignettes and help pages which works on the HTML site pkgdown builds). I also finished the Performance vignette, which talks about Bitset operations, prefabs, and writing new C++ functions.

File: vignettes/API.Rmd [hunk 2]
```diff
@@ -18,18 +18,18 @@ knitr::opts_chunk$set(
...
+<!--   1. [Introduction](#intro) -->
+<!--   2. [Variables](#var) -->
+<!--       1. [Categorical Variables](#cat_var) -->
+<!--       2. [Integer Variables](#int_var) -->
+<!--       3. [Double Variables](#dbl_var) -->
+<!--   3. [Processes](#proc) -->
+<!--       1. [Render](#rend) -->
+<!--   4. [Events](#event) -->
+<!--       1. [Targeted Events](#t_event) -->
+<!--   5. [Simulate](#sim) -->
```
Comment (giovannic) [inline vignettes/API.Rmd:32]: I found [this](https://github.com/r-lib/pkgdown/commit/bc69d25cd908d99e59208def4f111514058fafa1)

Perhaps something like:
```suggestion
```{r conditional_block, eval=!pkgdown::is_pkgdown()}
print("## Table of Contents {#toc}

  1. [Introduction](#intro)
  2. [Variables](#var)
      1. [Categorical Variables](#cat_var)
      2. [Integer Variables](#int_var)
      3. [Double Variables](#dbl_var)
  3. [Processes](#proc)
      1. [Render](#rend)
  4. [Events](#event)
      1. [Targeted Events](#t_event)
  5. [Simulate](#sim)")
  \\```
```

Comment (slwu89) [reply vignettes/API.Rmd]: @giovannic thanks! I added conditional TOCs to Tutorial, Performance, and API. Looks like this:

```
```{r conditional_block, eval=!pkgdown::in_pkgdown(),echo=F}
knitr::asis_output(
"## Table of Contents {#toc}

  1. [Introduction](#intro)
  2. [Bitset](#bitset)
  3. [Prefabs](#prefab)
  4. [C++ Prefabs](#cpp_prefab)"
)
```
```

File: vignettes/API.Rmd [hunk 3]
```diff
@@ -18,19 +18,23 @@ knitr::opts_chunk$set(
...
-<!--       1. [Categorical Variables](#cat_var) -->
-<!--       2. [Integer Variables](#int_var) -->
-<!--       3. [Double Variables](#dbl_var) -->
-<!--   3. [Processes](#proc) -->
-<!--       1. [Render](#rend) -->
-<!--   4. [Events](#event) -->
-<!--       1. [Targeted Events](#t_event) -->
-<!--   5. [Simulate](#sim) -->
-
+```{r conditional_block, eval=!pkgdown::in_pkgdown(),echo=F}
```
Comment (slwu89) [inline vignettes/API.Rmd:21]: @giovannic thanks for finding that conditional evaluation! combined with `knitr::asis_output` we get the right behavior (no TOC in the pkgdown site, yes TOC in the HTML vignettes).

**Review by giovannic — APPROVED** [review]

**Pull request merged**